Shell/main.cpp

```diff
@@ -147,6 +147,35 @@ static int sh_unset(int argc, char** argv)
     return 0;
 }
 
+static String expanded_tilde_expression_from(char* arg)
+{
+    int len = strlen(arg);
+    ASSERT(len);
+
+    StringBuilder login_name;
+    int first_slash_index = len;
+    for (int i = 1; i < len; ++i) {
+        if (arg[i] == '/') {
+            first_slash_index = i;
+            break;
+        }
+        login_name.append(arg[i]);
+    }
+
+    StringBuilder path;
+    for (int i = first_slash_index; i < len; ++i)
+        path.append(arg[i]);
+
+    if (login_name.length() == 0) {
+        char* home = getenv("HOME");
+        if (!home)
+            return {};
+        return String::format("%s/%s", home, path.to_string().characters());
+    }
+
+    return String::format("/home/%s/%s", login_name.to_string().characters(), path.to_string().characters());
```

Comment:
Let's use `getpwnam(login_name.characters())` instead of assuming that home directories are always `/home/username/`